TestYamllsDiagnosticsIntegrationWithSchema fails

`go test` fails, this is on Linux 6.14.x, with go 1.24.3.

```
--- FAIL: TestYamllsDiagnosticsIntegrationWithSchema (1.16s)
    diagnostics_integration_test.go:168:
                Error Trace:    github.com/mrjosh/helm-ls/internal/adapter/yamlls/diagnostics_integration_test.go:168
                Error:          []protocol.Diagnostic{protocol.Diagnostic{Range:protocol.Range{Start:protocol.Position{Line:0x1, Character:0x0}, End:protocol.Position{Line:0x1, Character:0x5}}, Severity:1, Code:513, CodeDescription:(*protocol.CodeDescription)(nil), Source:"yaml-schema: https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.32.1-standalone-strict/_definitions.json", Message:"Yamlls: Property wrong is not allowed.", Tags:[]protocol.DiagnosticTag(nil), RelatedInformation:[]protocol.DiagnosticRelatedInformation(nil), Data:map[string]interface {}{"properties":[]interface {}{"status"}, "schemaUri":[]interface {}{"https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.32.1-standalone-strict/_definitions.json"}}}} does not contain protocol.Diagnostic{Range:protocol.Range{Start:protocol.Position{Line:0x1, Character:0x0}, End:protocol.Position{Line:0x1, Character:0x5}}, Severity:1, Code:0, CodeDescription:(*protocol.CodeDescription)(nil), Source:"yaml-schema: https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.22.4-standalone-strict/_definitions.json", Message:"Yamlls: Property wrong is not allowed.", Tags:[]protocol.DiagnosticTag(nil), RelatedInformation:[]protocol.DiagnosticRelatedInformation(nil), Data:map[string]interface {}{"properties":[]interface {}{"status"}, "schemaUri":[]interface {}{"https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.22.4-standalone-strict/_definitions.json"}}}
                Test:           TestYamllsDiagnosticsIntegrationWithSchema
FAIL
FAIL    github.com/mrjosh/helm-ls/internal/adapter/yamlls       11.474s
ok      github.com/mrjosh/helm-ls/internal/charts       0.022s
?       github.com/mrjosh/helm-ls/internal/documentation/godocs [no test files]
?       github.com/mrjosh/helm-ls/internal/documentation/helm   [no test files]
ok      github.com/mrjosh/helm-ls/internal/handler      0.510s
ok      github.com/mrjosh/helm-ls/internal/helm_lint    0.019s
ok      github.com/mrjosh/helm-ls/internal/language_features    0.008s
?       github.com/mrjosh/helm-ls/internal/log  [no test files]
ok      github.com/mrjosh/helm-ls/internal/lsp  0.014s
ok      github.com/mrjosh/helm-ls/internal/protocol     0.003s
ok      github.com/mrjosh/helm-ls/internal/tree-sitter/gotemplate       0.002s
ok      github.com/mrjosh/helm-ls/internal/util 0.005s
?       github.com/mrjosh/helm-ls/internal/version      [no test files]
?       github.com/mrjosh/helm-ls/mocks/go.lsp.dev/protocol     [no test files]
```

--- a/Makefile
+++ b/Makefile
@@ -60,7 +60,7 @@ install-testrunner:
 	@$(GO) install -v gotest.tools/gotestsum@latest
 
 install-yamlls:
-	npm install --global yaml-language-server@1.15
+	npm install --global yaml-language-server@1.17
 
 integration-test-deps:
 	@YAMLLS_BIN=$$(command -v yaml-language-server) || { echo "yaml-language-server command not found! Installing..." && $(MAKE) install-yamlls; };

--- a/internal/adapter/yamlls/diagnostics_integration_test.go
+++ b/internal/adapter/yamlls/diagnostics_integration_test.go
@@ -23,7 +23,7 @@ import (
 var TEST_DATA_DIR = "../../../testdata/charts/bitnami/"
 
 func readTestFiles(dir string, channel chan<- string, doneChan chan<- int) {
-	libRegEx, e := regexp.Compile(".*(/|\\\\)templates(/|\\\\).*\\.ya?ml")
+	libRegEx, e := regexp.Compile(`.*(/|\\)templates(/|\\).*\.ya?ml`)
 	if e != nil {
 		log.Fatal(e)
 		return
@@ -116,7 +116,6 @@ func TestYamllsDiagnosticsIntegration(t *testing.T) {
 
 func TestYamllsDiagnosticsIntegrationWithSchema(t *testing.T) {
 	t.Parallel()
-	diagnosticsChan := make(chan lsp.PublishDiagnosticsParams)
 
 	config := util.DefaultConfig.YamllsConfiguration
 	yamllsConnector, documents, diagnosticsChan := getYamlLsConnector(t, config)
@@ -137,32 +136,29 @@ func TestYamllsDiagnosticsIntegrationWithSchema(t *testing.T) {
 		Severity:           1,
 		Code:               0.0,
 		CodeDescription:    nil,
-		Source:             "yaml-schema: https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.22.4-standalone-strict/_definitions.json",
+		Source:             "yaml-schema: https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/MOCKED-VERSION-standalone-strict/_definitions.json",
 		Message:            "Yamlls: Property wrong is not allowed.",
 		Tags:               nil,
 		RelatedInformation: nil,
-		Data: map[string]interface{}{
-			"properties": []interface{}{
-				"status",
-			},
-			"schemaUri": []interface{}{
-				"https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.22.4-standalone-strict/_definitions.json",
-			},
-		},
+		Data:               map[string]any{},
 	}
 
 	diagnostic := []lsp.Diagnostic{}
 	afterCh := time.After(10 * time.Second)
-	for {
-		if len(diagnostic) > 0 {
-			break
-		}
+	for len(diagnostic) <= 0 {
 		select {
 		case d := <-diagnosticsChan:
 			diagnostic = append(diagnostic, d.Diagnostics...)
 		case <-afterCh:
 			t.Fatal("Timed out waiting for diagnostics")
 		}
+	}
+
+	regex := regexp.MustCompile(`/v\d+\.\d+\.\d+-standalone-strict/`)
+
+	for i := range diagnostic {
+		diagnostic[i].Source = regex.ReplaceAllString(diagnostic[i].Source, "/MOCKED-VERSION-standalone-strict/")
+		diagnostic[i].Data = map[string]any{}
 	}
 
 	assert.Contains(t, diagnostic, expected)

--- a/internal/adapter/yamlls/diagnostics_integration_test.go
+++ b/internal/adapter/yamlls/diagnostics_integration_test.go
@@ -134,7 +134,7 @@ func TestYamllsDiagnosticsIntegrationWithSchema(t *testing.T) {
 			},
 		},
 		Severity:           1,
-		Code:               0.0,
+		Code:               0,
 		CodeDescription:    nil,
 		Source:             "yaml-schema: https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/MOCKED-VERSION-standalone-strict/_definitions.json",
 		Message:            "Yamlls: Property wrong is not allowed.",
@@ -159,6 +159,7 @@ func TestYamllsDiagnosticsIntegrationWithSchema(t *testing.T) {
 	for i := range diagnostic {
 		diagnostic[i].Source = regex.ReplaceAllString(diagnostic[i].Source, "/MOCKED-VERSION-standalone-strict/")
 		diagnostic[i].Data = map[string]any{}
+		diagnostic[i].Code = 0
 	}
 
 	assert.Contains(t, diagnostic, expected)

